source/material_model/rheology/visco_plastic.cc

```diff
@@ -315,6 +321,14 @@ namespace aspect
                 }
               }
 
+            // TODO: Implement fixed-point iterations. 
+            // Explanation: The variables in Eq. 36 of Moresi et al. (2003) for the effective viscosity
+            // (i.e. yield stress, viscous viscosity, shear modulus, elastic timestep and lambda) are not
+            // likely to be constant within a timestep, as through the nonlinear iterations the viscous
+            // viscosity (strain rate and pressure dependency) and the yield stress (pressure dependency)
+            // are likely to be updated. In this case, local iterations between the viscoelastic stress
+            // and the plastic stress need to be performed to find lambda.
```

Comment:
I think I misspoke over zoom. As long as none of these variables are strain-dependent, it shouldn't be necessary to do any local iterations - the nonlinear iterations should already deal with pressure dependency, for example. 

The main issue is that the code does allow for strain dependent viscosity (i.e. the dislocation creep and Peierls options), and in these cases we need to calculate the stress state of the system, and calculate the viscosities using that stress.

Reply:
Okay, I'll update that. But that means we should implement the local iterations.